src/model.py

```diff
@@ -2002,79 +1969,62 @@ def init_components(self, subtasks):
-                elif None not in (split_cnn_tasks, split_cnn_kernels): 
-                    if task in split_cnn_tasks:
-                        components[task].update({
-                            f"cnn_{stack}": self.split_cnn_block(
-                                cnn_dim, cnn_dim, split_cnn_kernels, task_layers
-                                )
-                            })
-                    else:
-                        components[task].update(
-                            {f"cnn_{stack}": torch.nn.Sequential(*[
-                                self.cnn_block(cnn_dim, cnn_dim, kernel_size)
-                                for layer in range(task_layers)
-                                ])}
+                        ])
+                    elif None not in (split_cnn_tasks, split_cnn_kernels): 
+                        if task in split_cnn_tasks:
+                            components[task][f"cnn_{stack}"] = self.split_cnn_block(
+                                cnn_dim*2, cnn_dim, split_cnn_kernels, task_layers
                             )
-
-                else:
-                    components[task].update(
-                        {f"cnn_{stack}": torch.nn.Sequential(*[
-                            self.cnn_block(cnn_dim, cnn_dim, kernel_size)
+                        else:
+                            components[task][f"cnn_{stack}"] = torch.nn.Sequential(*[
+                                self.cnn_block(cnn_dim*2, cnn_dim, kernel_size)
+                                for layer in range(task_layers)
+                            ])
+                    else:
+                        components[task][f"cnn_{stack}"] = torch.nn.Sequential(*[
+                            self.cnn_block(cnn_dim*2, cnn_dim, kernel_size)
                             for layer in range(task_layers)
-                            ])}
-                        )
-                
+                        ])
+        
+                    ## Feedforward component: Expands per stack due to concatentation
+                    components[task][f"linear_{stack}"] = self.linear_block(in_features=(cnn_dim*2), out_features=3)
 
+        else: 
+            for task in subtasks:
+                # final outputs for each task
+                # in_features are concats of embeddings, shared layers, and task cnns
+                components[task]["linear"] = self.linear_block(in_features=cnn_dim*2, out_features=3)
         ######################################
         # Shared relation components
         ######################################
 
-        # similar to IMN setup
-        # NOTE main difference: IMN uses split_cnn_block w/ kernel=3 & kernel=5
-        if split_cnn_kernels is not None:  # NOTE only split on first shared
+        ## Shared layers: similar to IMN setup
+        # default difference: IMN uses split_cnn_block w/ kernel=3 & kernel=5
+        if split_cnn_kernels is not None:  # NOTE only splitting on first shared
```

Comment:
Only splitting on first shared cnn bc thats how its done in IMN.

Future work can look into implementing this for all shared, if that could possible give even better performance.